src/bp/lang-server/service/index.ts

```diff
@@ -45,75 +54,172 @@ export default class LanguageService {
     return this._ready
   }
 
-  listModels(): AvailableModel[] {
-    return _.values(this._models)
+  listFastTextModels(): AvailableModel[] {
+    const allModels = _.values(this._models)
+    return allModels.map(m => m.fastTextModel)
   }
 
   private _discoverModels() {
     if (!fs.existsSync(this.langDir)) {
-      return []
+      return
     }
 
     const files = fs.readdirSync(this.langDir)
-    return files.forEach(f => {
+
+    type modelFileInfo = {
+      domain: string
+      langCode: string
+      file: string
+      dim?: number
+    }
+    const filesInfo: modelFileInfo[] = []
+
+    files.forEach(f => {
       // Examples:  "scope.en.300.bin" "bp.fr.150.bin"
-      const regex = /(\w+)\.(\w+)\.(\d+)\.bin/i
-      const match = f.match(regex)
-      if (!match) {
+      const fastTextModelsRegex = /(\w+)\.(\w+)\.(\d+)\.bin/i
+      const fastTextModelsMatch = f.match(fastTextModelsRegex)
+
+      // Examples: "scope.en.bpe.model" "bp.en.bpe.model"
+      const bpeModelsRegex = /(\w+)\.(\w+)\.bpe\.model/i
+      const bpeModelsMatch = f.match(bpeModelsRegex)
+
+      if (!!fastTextModelsMatch) {
```

Comment:
The !! aren't required here, since if there is no match, it will already be null. 

We use mostly !! when you are using the condition for "length" in a "ui" context, for ex, if you write  <div>{models.length && models[0].something}</div>, if there is 0 models, it will display 0, but if you have one or more model, it will just display models[0].something. 

In that case, we would use {!!models.length &&...} so it's converted in a boolean.